google/cloud/bigtable/internal/table_admin.cc

```diff
@@ -260,11 +259,11 @@ void TableAdmin::ListSnapshotsImpl(
       break;
     }
 
-    for (auto& x : *response.mutable_snapshots()) {
-      inserter(x);
-    }
+    auto& snapshots = *response.mutable_snapshots();
+    std::move(snapshots.begin(), snapshots.end(), std::back_inserter(result));
     page_token = std::move(*response.mutable_next_page_token());
   } while (!page_token.empty());
+  return result;
```

Comment:
It's a little funny that this may return an empty vector (or even a non-empty vector?) even if the out-status is non-OK. It's up to the caller to ignore the returned vector based on the value of status. This seems like a case where we'd want to return `StatusOr<vector<Foo>>` which would cover both cases. 

Now I'm new to this bigtable library, so I'm not sure if we want to insert StatusOr in this internal interface... but I'm thinking maybe we do?  Thoughts?

Reply:
Sorry to jump in.
@coryan  asked me not to change the internal interface yet. I am working on #1965 and
I am using this issue as an example for my changes.

Reply:
@rsimion Did I?  I probably did, but it all depends on context.  @devjgm introducing `StatusOr<>` here is probably Okay, in any case this class (`bigtable::noex::TableAdmin`) should be removed soon after this PR is merged because then `bigtable::TableAdmin` would have identical functionality and a cleaner interface.